src/openrct2/world/Map.cpp

```diff
@@ -313,16 +313,16 @@ int32_t TileElementIteratorNext(TileElementIterator* it)
         return 1;
     }
 
-    if (it->y < (MAXIMUM_MAP_SIZE_TECHNICAL - 1))
+    if (it->y < (GetGameState().MapSize.y - 2))
```

Comment:
You could create a reference to the gameState to avoid fetching it twice.

```cpp
auto& gameState = GetGameState();
```

Reply:
Implemented this.